### ISSUE TITLE
React Legacy Context API deprecation.

### ISSUE THREAD

**Issue by radiantshaw** [danott/react-rails-form-helpers#25]
Hello @danott 

Great work on this package as this was exactly what we were looking for for our use case. I just have one question. Since this package uses the legacy context API, which React is now working towards removal of in the future major version, are there any plans to convert this to use the new context API instead?

**Comment by danott** [comment]
Absolutely. Issue #17 is where I first noted the need to modernize to current React idioms. With official plans to remove the legacy context API in the next major version, it sounds like we're transitioning from a quality-of-life change to a necessary-for-survival change.

**Comment by danott** [comment]
I am spring cleaning old issues.
